src/loaders/zarr_utils/utils.ts

```diff
@@ -101,3 +109,130 @@ export function getScale(dataset: OMEDataset | OMEMultiscale, orderTCZYX: TCZYX<
   const scale = scaleTransform.scale.slice();
   return orderByTCZYX(scale, orderTCZYX, 1);
 }
+
+/**
+ * Defines a partial order of zarr arrays based on their size. Specifically:
+ * - If array size x, y, z are all equal, the arrays are equal
+ * - otherwise, if all xyz of `a` are less than or equal to those of `b`, `a` is less than `b` (and vice versa)
+ * - if some xyz is less and some is greater, the arrays are uncomparable
+ */
+function compareZarrArraySize(
+  aArr: NumericZarrArray,
+  aTCZYX: TCZYX<number>,
+  bArr: NumericZarrArray,
+  bTCZYX: TCZYX<number>
+): number | undefined {
+  const diffZ = aArr.shape[aTCZYX[2]] - bArr.shape[bTCZYX[2]];
+  const diffY = aArr.shape[aTCZYX[3]] - bArr.shape[bTCZYX[3]];
+  const aX = aTCZYX[4] > -1 ? aArr.shape[aTCZYX[4]] : 1;
+  const bX = bTCZYX[4] > -1 ? bArr.shape[bTCZYX[4]] : 1;
+  const diffX = aX - bX;
+
+  if (diffZ === 0 && diffY === 0 && diffX === 0) {
+    return 0;
+  } else if (diffZ <= 0 && diffY <= 0 && diffX <= 0) {
+    return -1;
+  } else if (diffZ >= 0 && diffY >= 0 && diffX >= 0) {
+    return 1;
+  } else {
+    return undefined;
+  }
+}
+
+function scaleTransformsAreEqual(aSrc: ZarrSource, aLevel: number, bSrc: ZarrSource, bLevel: number): boolean {
+  const aScale = getScale(aSrc.multiscaleMetadata.datasets[aLevel], aSrc.axesTCZYX);
+  const bScale = getScale(bSrc.multiscaleMetadata.datasets[bLevel], bSrc.axesTCZYX);
+  return aScale[2] === bScale[2] && aScale[3] === bScale[3] && aScale[4] === bScale[4];
```

Comment:
Nit: would this comparison benefit from a small epsilon?  When the same-shape multiresolution levels are offset then the scale values could be unequal.

Reply:
Good call! It probably would.